### DISCUSSION TITLE
Iterator + 'static

### DISCUSSION THREAD

**Discussion by wiseman** [dpc/pariter#7] [Q&A]
I love this crate, it's exactly what I need and I use it frequently.  Thank you!

It seems like you can't use the unscoped methods unless you're iterating over `static` values; is that right?  That's fine, but I just want to make sure I'm not confused. The README doesn't mention that limitation, but does talk about e.g using `.cloned()` to help with the issue of borrowed references, which I've tried but there's still the issue of e.g. `parallel_map` requiring `Iterator + 'static`.

**Comment by dpc** [comment]
Hmm. My lifetime fu is not expoert, but my understanding is that non-`'static` items imply holding on to something on the current thread which is not safe to send without scoped threads.

**Comment by safinaskar** [comment]
Here is possible solution: https://github.com/dpc/pariter/pull/9

**Comment by safinaskar** [comment]
Here is my patch https://github.com/dpc/pariter/pull/11 , which sometimes removes `'static` bounds
